cheatsheets_draft/Webhook_Security_Guidelines_CheatSheet.md

```diff
@@ -0,0 +1,169 @@
+# Webhook Security Guidelines Cheat Sheet
+
+## Introduction
+
+Webhooks provide a mechanism where a Server-side application can notify a client-side application when a new even (that the client-side application might be interested in) has occurred on the server. Webhooks are incredibly useful and a resource-light way to implement event reactions. However, webhooks can also be abused and the following considerations must be taken into account to secure Webhooks. It should also be noted that webhook security considerations apply to both publisher/server and subscriber/client. 
+
+At the core of webhooks fundamental security best practices are available however are not enabled by default. This is a gap in the “default secure” principle however with focus and configuration this gap can be addressed. One item to note is webhook urls by default carry a high level of confidentiality. This comes from the lack of default secure principle mentioned above. With this webhook urls should be kept secret and while the security community has generally accepted that obscurity is not security it is a best practice to follow. If your webhooks are sharing sensitive information the following considerations should be reviewed and implemented.
+
+At high level, the following requirements outline the security mechanisms in various ways. These high level requirements allow engineers and developers to properly implement security controls. 
+
+
+- **Confidentiality: A passive attacker must not be able to read Webhook Notifications, as they may contain confidential data.**
+- **Integrity: An active attacker must not be able to alter Webhook Notifications, as this may result in unexpected behaviour affecting your infrastructure.**
+- **Authenticity: An active attacker must not be able to forge Webhook Notifications as this may result in your receive code taking unwanted action.**
+
+
+## Contents
+
+Controls for Securing Webhooks
+
+- Securing Webhook URLs
+-	Enable SSL/TLS
+-	Basic Authentication
+- Mutual TLS Authentication
+-	Oath Authorization
+-	Crude Authorized Access
+-	Message Protection and Secrecy
+-	Cross Site Request Forgery
+-	Rotation and Retry
+-	Pass Secrets in Request header
+-	Fail Safe and Secure
+-	Replay Protections
+-	Payload best practices
+-	HTTP Methods best practices
+
+
+## Control 1 - Securing Webhook URLs
+
+Secure URLs - Keep URLs out of source code, keep urls out of logs. Do not print Webhook URLs in logs, Do not store write URLs or hard code it in source. Webhook URLs should be securely stored in Vault. 
```

Comment:
Sure, will make it as Unordered list. 

This is excellent reference where Researcher found over 6000 webhooks in github public repos. I believe this will help Developers to understand risk before they Webhooks in github public repos. I would need your help to put this in right place in cheatsheet. 
https://github.com/Gruppio/SlackWebhooksGithubCrawler

Reply:
When you push your new commits, we can word this inside of the control and make it link to this. Sure thing.

Reply:
The problem here is not that we should keep URL secret but to keep tokens secret - in the linked example slack includes authentication tokens as part of the URL. IMO this point should be part of bigger Authentication point because if you add on top of it any authentication then there is no need to keep it secret.